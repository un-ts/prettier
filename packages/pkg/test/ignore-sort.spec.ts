import { format } from 'prettier'

import pkg1 from './fixtures/fixture1.json'

import PkgPlugin from 'prettier-plugin-pkg'

test('ignore-sort', async () => {
  const input = JSON.stringify(pkg1, null, 2)
  const output = await format(input, {
    filepath: 'package.json',
    parser: 'json-stringify',
    plugins: [PkgPlugin],
    packageIgnoreSort: ['resolutions'],
  })

  expect(output).toMatchSnapshot()
})
