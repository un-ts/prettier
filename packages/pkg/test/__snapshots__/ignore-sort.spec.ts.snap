// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ignore-sort 1`] = `
"{
  "name": "prettier-plugin-pkg",
  "version": "0.0.0",
  "description": "An opinionated package.json formatter plugin for Prettier",
  "keywords": [
    "package",
    "package.json",
    "pkg",
    "plugin",
    "prettier",
    "prettier-plugin"
  ],
  "homepage": "https://github.com/un-ts/prettier/tree/master/packages/pkg",
  "license": "MPL-2.0",
  "author": "JounQin <admin@1stg.me> (https://www.1stG.me)",
  "files": [
    "lib",
    "rules"
  ],
  "main": "lib/cjs",
  "repository": "git@github.com/un-ts/prettier.git",
  "peerDependencies": {
    "prettier": "^1.18.2"
  },
  "engines": {
    "node": ">= 8.0.0"
  },
  "commitlint": {
    "extends": [
      "@1stg"
    ]
  },
  "es2015": "lib/es2015",
  "eslintIgnore": [
    "CHANGELOG.md",
    "lib",
    "!.*.js"
  ],
  "module": "lib/esm",
  "resolutions": {
    "imagemin-gifsicle": "^6.0.1",
    "@babel/core": "^7.9.0",
    "typescript": "^3.8.3",
    "prettier": "^2.0.4",
    "@babel/preset-env": "^7.12.1",
    "eslint-plugin-prettier": "^3.1.2"
  }
}
"
`;
